SO3_XL_ToolChange_fixedLocation

```diff
@@ -1,21 +1,39 @@
-;This macro allows you to use a fixed machine location for a tool change/probe. Ideal when you're workpiece surface has been carved away.
+; This macro allows you to use a fixed machine location for a tool change and probe. It works by
+; probing twice, once to capture the position of the tool you're changing away from, and again to
+; set the adjusted WCS offset of the tool you're changing to.
+
+; Using the macro in practice:
+; 1) Insert an M6 tool change command in your program, which will pause cncjs and allow you to
+;     execute this macro.
+; 2) The router retracts to a safe height and moves to a pre-defined location for tool changes and
+;     probing, then pauses.
+; 3) Turn off your router, set up your probe, and continue. This will probe to capture the position
+;     of your current tool and then retract back up to the tool change height and pause.
+; 4) Change out your tool, set up your probe, and continue. This will probe the new tool. When it
+;     reaches the probe, the end of the tool will be in precisely the same position as the original
+;     tool, but the Z position will be different, so we set the current Z offset to be the previously
+;     captured offset.
+; 5) Remove the probe and continue. The spindle will move back to the location it encountered
+;     the M6.
+; 6) Turn your router back on and resume execution of your program.
 
-; Wait until the planner queue is empty
 %wait
-; Set user-defined variables
-%SAFE_HEIGHT = -5	;clear everything height (negative number, distance below Z limit)
-;Following set probe location
-%TOOL_PROBE_X = -30	;machine coordinates
-%TOOL_PROBE_Y = -350	;machine coordinates
-%TOOL_PROBE_Z = -20	;machine coordinates --> lower this (more negative) to start the probing closer to wasteboard
+G21 ; Metric
 
+; User-defined variables (***METRIC***)
+%SAFE_HEIGHT = -1 ; Height needed to clear everything (negative number, distance below Z limit)
+%TOOL_PROBE_X = -30 ; Machine coords
+%TOOL_PROBE_Y = -350 ; Machine coords
+%TOOL_PROBE_Z = -20   ; Machine coords
 %PROBE_DISTANCE = 100
-%PROBE_FEEDRATE = 75
-%RETRACTION_DISTANCE = 10
+%PROBE_FAST_FEEDRATE = 200 ; mm/min
+%PROBE_SLOW_FEEDRATE = 20 ; mm/min
```

Comment:
I think 20 mm/min is too slow for an SO3.
I know it's user definable, but we want defaults to be reasonable for the common use.